### ISSUE TITLE
New Error: Could not connect to Sauce Labs API: Error: read ECONNRESET

### ISSUE THREAD

**Issue by LaurentGoderre** [axemclion/grunt-saucelabs#174]
All our builds are failing with the following error now

```
Could not connect to Sauce Labs API: Error: read ECONNRESET
```

I am not quite sure where the error is coming from yet

**Comment by endel** [comment]
I'm having this problem too. Is it a server-side problem?

**Comment by Jonahss** [comment]
I'm going on the assumption that it is a server-side problem, and we're trying to nail down a simpler reproduction case.

**Comment by LaurentGoderre** [comment]
Trying to narrow it down too. Will let you know what I find.

**Comment by Jonahss** [comment]
^.^

**Comment by Jonahss** [comment]
Is everyone who got this error using Sauce Connect tunnels?

**Comment by cvrebert** [comment]
I use tunnels and got this error.

**Comment by LaurentGoderre** [comment]
Same here

**Comment by msukmanowsky** [comment]
Using Sauce Connect and also getting this error. Does seem to hit me when running >= 3 browsers or so. SauceLabs reporting `The Sauce VMs failed to start the browser or device For more info, please check https://docs.saucelabs.com/reference/troubleshooting-common-error-messages/` in the interface despite the browsers I'm using being fully supported:

```json
[
    {
        "browserName": "firefox",
        "platform": "Windows 8.1",
        "version": "15",
        "deviceName": ""
    },
    {
        "browserName": "chrome",
        "platform": "Windows 8.1",
        "version": "26",
        "deviceName": ""
    },
    {
        "browserName": "internet explorer",
        "platform": "Windows 8.1",
        "version": "11",
        "deviceName": ""
    },
    {
        "browserName": "firefox",
        "platform": "OS X 10.6",
        "version": "15",
        "deviceName": ""
    }
]
```

**Comment by msukmanowsky** [comment]
Feels as though one test encountered `The Sauce VMs failed to start the browser or device For more info, please check https://docs.saucelabs.com/reference/troubleshooting-common-error-messages/` and the tunnel was immediately disconnected instead of allowing other tests to complete first.

**Comment by Jonahss** [comment]
Yeah, we should check the logic for closing tunnels, we should make sure the tunnel isn't closed until all tests have fully completed.

**Comment by LaurentGoderre** [comment]
Shouldn't that be a separate issue though?

**Comment by Jonahss** [comment]
The error is different, but the way we handle tunnels might be responsible
for the same thing we see here.

**Comment by LaurentGoderre** [comment]
Not sure, the suggested PR seems to be fixing this specific issue

**Comment by Jonahss** [comment]
See the discussion [here](#180). @gvas found that specific versions of Node.js cause issues.

**Comment by Jonahss** [comment]
This should be fixed in version 8.5.0


comment here if you still encounter ECONNRESET errors after updating.

**Comment by LaurentGoderre** [comment]
Thanks everyone for the help!!
